src/contexts/auth-provider.js

```diff
@@ -11,6 +11,13 @@ const AuthProvider = ({ children }) => {
   const [loaded, setLoaded] = useState(false);
   const [trigger, setTrigger] = useState(false);
 
+  const resetUserData = () => {
+    setTrigger(false);
+    setUser();
+    setIsAuthenticated(false);
+    setLoaded(false);
```

Comment:
I think removing this line solves the forever loading problem right after logging out. I believe the reason is because the useEffect that is listening to `trigger` is the one in charge of setting loaded to `true`. If we `setLoaded(false)` after the trigger, then loaded will remain as `false` (and the Loader will stay forever) until the user refreshes the page.